lib/src/main/java/com/marcinorlowski/bettertextinputlayout/BetterTextInputLayout.java

```diff
@@ -87,9 +98,7 @@ public void togglePassword() {
 	 */
 	protected void setToggleState(boolean passwordVisible) {
 		try {
-			Field f = this.getClass().getSuperclass().getDeclaredField("mPasswordToggleView");
-			f.setAccessible(true);
-			CheckableImageButton b = (CheckableImageButton)f.get(this);
+			CheckableImageButton b = (CheckableImageButton) passwordToggleViewField.get(this);

```

Comment:
If we cache `Method`, shouldn't we case `CheckableImageButton` instead of `mPasswordToggleView` as well?